pkg/core/native/policy.go

```diff
@@ -713,15 +806,170 @@ func (p *Policy) CheckPolicy(d *dao.Simple, tx *transaction.Transaction) error {
 	return nil
 }
 
-// iterator provides an iterator over a slice of keys.
-type iterator struct {
-	keys []util.Uint160
+// IsWhitelisted checks whether the specified contract method is whitelisted and
+// returns a non-negative execution fee in picoGAS units if so. It always uses
+// native cache, hence cache is expected to be initialized by this moment.
+func (p *Policy) IsWhitelisted(dao *dao.Simple, hash util.Uint160, offset int) int64 {
+	cache := dao.GetROCache(p.ID).(*PolicyCache)
+	i, ok := slices.BinarySearchFunc(cache.whitelistedContracts, whitelistedContract{
+		hash:   hash,
+		offset: uint32(offset),
+	}, whitelistedContract.Compare)
+	if !ok {
+		return -1
+	}
+	return cache.whitelistedContracts[i].fee * vm.ExecFeeFactorMultiplier
+}
+
+// CleanWhitelist removes a contract with the specified hash from the list of
+// whitelisted contracts.
+func (p *Policy) CleanWhitelist(ic *interop.Context, cs *state.Contract) error {
+	var (
+		cache = ic.DAO.GetRWCache(p.ID).(*PolicyCache)
+		err   error
+	)
+	cache.whitelistedContracts = slices.DeleteFunc(cache.whitelistedContracts, func(c whitelistedContract) bool {
+		if err != nil {
+			return false
+		}
+		if !c.hash.Equals(cs.Hash) {
+			return false
+		}
+		k := makeWhitelistedKey(c.hash, c.offset)
+		ic.DAO.DeleteStorageItem(p.ID, k)
+		m := cs.Manifest.ABI.GetMethodByOffset(int(c.offset))
```

Comment:
But method signature is name/arg count as in https://github.com/neo-project/neo/pull/4201/files#diff-1d4d178ba1fae37acb3548bf56809483baac60a5a6b5c1c8d4d88b45ca37adf8R365